sharding/notary/service.go

```diff
@@ -33,13 +28,13 @@ func (n *Notary) Start() error {
 	// TODO: handle this better through goroutines. Right now, these methods
 	// have their own nested channels and goroutines within them. We need
 	// to make this as flat as possible at the Notary layer.
-	if n.node.DepositFlagSet() {
-		if err := joinNotaryPool(n.node); err != nil {
+	if n.smcClient.DepositFlag() {
+		if err := joinNotaryPool(n.smcClient); err != nil {
 			return err
 		}
 	}
 
-	return subscribeBlockHeaders(n.node)
+	return subscribeBlockHeaders(n.smcClient)
```

Comment:
This needs to be non-blocking so that other services can be started.

Two options:
1) Add a TODO and do this later
2) Put this into a goroutine with some logic to cleanup in `Stop`

Reply:
I think this whole routine needs to be non-blocking. I think joinNotaryPool will make the client hang until its joined.

The idea is that we might have some stats services or something that are registered after the actors

Reply:
Added a TODO to move to a separate PR